hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -2164,6 +2064,124 @@ public FileStatus getFileStatus(Path path, TracingContext tracingContext, boolea
     }
   }
 
+  /**
+   * Method that deals with error cases of calling getPathProperty directly.
+   * getPathProperty itself does not do the error handling, as it is intended to be
+   * one part of the calls constituting getFileStatus. This additional method
+   * would help when getPathProperty has to be called in a direct flow and needs a check for this error.
+   * @param path Current Path
+   * @param tracingContext current tracing context
+   * @param useBlobEndpoint whether to use blob endpoint
+   * @return FileStatus or null if blob does not exist or is not explicit
+   * @throws IOException
+   */
+  FileStatus tryGetPathProperty(Path path, TracingContext tracingContext, Boolean useBlobEndpoint) throws IOException {
+    try {
+      return getPathProperty(path, tracingContext, useBlobEndpoint);
+    } catch(AbfsRestOperationException ex) {
+      if (ex.getStatusCode() == HTTP_NOT_FOUND) {
+        LOG.debug("No explicit directory/path found: {}", path);
+        return null;
+      }
+      throw ex;
+    }
+  }
+
+  /**
+   * Method to make a call to get path property based on various configs-
```

Comment:
We should specify in comments that this function will work only for explicit paths (files/directories).